app/features/quizzify/tools.py

```diff
@@ -83,26 +126,200 @@ def load(self) -> List[Document]:
+                raise ValueError(f"Unsupported file type: {file_type}")
+            
+        return documents
 
-                for i, page in enumerate(pdf_reader.pages):
-                    page_content = page.extract_text()
-                    metadata = {"source": file_type, "page_number": i + 1}
+class BytesFileXLSXLoader(BaseLoader):
 
-                    doc = Document(page_content=page_content, metadata=metadata)
+    def __init__(self, files: List[Tuple[BytesIO, str]]):
+        self.files = files
+    
+    def load(self) -> List[Document]:
+        documents = []
+        
+        for file, file_type in self.files:
+            logger.debug(file_type)
+            if file_type.lower() == "xlsx":
+                # pdf_reader = PdfReader(file) #! PyPDF2.PdfReader is deprecated
+                file.seek(0)
+                df = pd.read_excel(file)
+                for row in df.itertuples():
+                    content = ""
+                    for column in row[1:]:
+                        content+= (str(column).strip() + "\n")
+                    metadata = {"page_number": row[0] + 1, "source": file_type}
+                    doc = Document(page_content=content, metadata=metadata)
+                    documents.append(doc)               
+            else:
+                raise ValueError(f"Unsupported file type: {file_type}")
+            
+        return documents
+     
+class DocLoader(BaseLoader):
+
+    def __init__(self, files: List[Tuple[BytesIO, str]]):
+        self.files = files
+    
+    def load(self) -> List[Document]:
+        documents = []
+        
+        for file, file_type in self.files:
+            logger.debug(file_type)
+            if file_type.lower() == "docx":
+                # pdf_reader = PdfReader(file) #! PyPDF2.PdfReader is deprecated
+                docs = docu(file)
+                for page_num, page in enumerate(docs.paragraphs):
+                    page_content = ""
+                    for paragraph in page.runs:
+                        page_content += paragraph.text.strip() + "\n"
+                    metadata = {"page_number": page_num + 1, "source": file_type}
+                    doc = Document(page_content=page_content.rstrip(),metadata=metadata)
+                    documents.append(doc)               
+            else:
+                raise ValueError(f"Unsupported file type: {file_type}")
+            
+        return documents
+    
+class ImageLoader(BaseLoader):
+    def __init__(self,files: List[Tuple[BytesIO,str]]):
+        self.files = files
+    
+    def load(self) -> List[Document]:
+        documents = []
+        text_completion_model = VertexAI(model='gemini-1.5-flash-001')
+        prompt= PromptTemplate.from_template("I want you to check if there is any missing words in {text}. If there are any, I want to to autocomplete them with the most relevant word possible and make the whole thing grammatically correct. The output should be a string.")
```

Comment:
There are better prompting techniques. You can gain inspiration from:https://www.promptingguide.ai/techniques